src/UI/Tables/Stateable.elm

```diff
@@ -0,0 +1,746 @@
+        |> Filters.SingleTextFilter
+        |> swap filtersPush accu
+
+
+filterRemote : msgs -> Strategy msgs value item
+filterRemote msgs =
+    Filters.Remote msgs
+
+
+
+-- Width
+
+
+{-| Applies [`Element.width`](/packages/mdgriffith/elm-ui/latest/Element#width) to the component.
+
+    Table.withWidth
+        (Element.fill |> Element.minimum 220)
+        someTable
+
+-}
+withWidth : Element.Length -> StateableTable msg item columns -> StateableTable msg item columns
+withWidth width (Table prop opt_) =
+    Table prop { opt_ | width = width }
+
+
+
+-- Rendering
+
+
+{-| End of the builder's life.
+The result of this function is a ready-to-insert Elm UI's Element.
+-}
+renderElement : RenderConfig -> StateableTable msg item columns -> Element msg
+renderElement renderConfig (Table prop opt) =
+    case opt.responsive of
+        Just responsive ->
+            if RenderConfig.isMobile renderConfig then
+                mobileView renderConfig prop opt responsive
+
+            else
+                desktopView renderConfig prop opt
+
+        Nothing ->
+            desktopView renderConfig prop opt
+
+
+
+-- Mobile rendering
+
+
+mobileView :
+    RenderConfig
+    -> StateableConfig msg item columns
+    -> Options msg item columns
+    -> Responsive msg item columns
+    -> Element msg
+mobileView renderConfig prop opt responsive =
+    ListView.toggleableList
+        { detailsShowLabel = "Expand"
+        , detailsCollapseLabel = "Collapse"
+        , toCover = Tuple.second >> responsive.toCover
+        , toDetails =
+            Tuple.second
+                >> responsive.toDetails
+                >> NArray.toList
+                >> List.filterMap (Maybe.map (detailView renderConfig))
+        , selectMsg = Tuple.first >> MobileToggle >> prop.toExtern
+        }
+        |> ListView.withItems (List.indexedMap Tuple.pair opt.items)
+        |> ListView.withSelected (Tuple.first >> isSelected prop.state)
+        |> ListView.renderElement renderConfig
+
+
+isSelected : State -> Int -> Bool
+isSelected (State { mobileSelected }) position =
+    Just position == mobileSelected
+
+
+detailView : RenderConfig -> Detail msg -> ( String, Element msg )
+detailView renderConfig { label, content } =
+    ( label, cellContentRender renderConfig content )
+
+
+
+-- Dekstop rendering
+
+
+desktopView :
+    RenderConfig
+    -> StateableConfig msg item columns
+    -> Options msg item columns
+    -> Element msg
+desktopView renderConfig prop opt =
+    let
+        len =
+            NArray.length prop.columns
+
+        indexedList =
+            List.range 0 (len - 1)
+
+        columns =
+            NArray.toList prop.columns
+
+        filters =
+            opt.filters
+                |> Maybe.map (NArray.map Just >> NArray.toList)
+                |> Maybe.withDefault (List.repeat len Nothing)
+
+        filtersState =
+            case prop.state of
+                State state ->
+                    List.map
```

Comment:
You could use List.indexedMap fn opt.filters this way I think `len` and `indexedList` is not required.